chain/client-primitives/src/types.rs

```diff
@@ -264,7 +264,42 @@ impl Query {
 }
 
 impl Message for Query {
-    type Result = Result<Option<QueryResponse>, String>;
+    type Result = Result<QueryResponse, QueryError>;
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum QueryError {
+    #[error("IO Error: {0}")]
+    IOError(String),
+    #[error("There are no fully synchronized blocks yet")]
+    NotSyncedYet,
+    #[error("The node does not track the shard")]
+    DoesNotTrackShard,
+    #[error("Invalid account ID {0}")]
+    InvalidAccount(near_primitives::types::AccountId),
+    #[error("Account ID {0} has never been observed on the node")]
+    AccountDoesNotExist(near_primitives::types::AccountId),
+    #[error("Contract code for contract ID {0} has never been observed on the node")]
+    ContractCodeDoesNotExist(near_primitives::types::AccountId),
+    #[error("Access key for public key {0} has never been observed on the node")]
+    AccessKeyDoesNotExist(String),
+    #[error("VM error occurred: {0}")]
+    VMError(String),
```

Comment:
just a reminder to align these accordingly to the comment I left to the enum in jsonrpc-primitives